### ISSUE TITLE
[$250] Xero - "Taxes" in the workspace settings are only appearing after a browser refresh

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54155]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.76-6
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5343677&group_by=cases:section_id&group_id=309134&group_order=asc
**Issue reported by:** Applause Internal Team
**Device used:** Windows 10 / Chrome, Apple iPhone 12 Pro / Safari
**App Component:** Workspace Settings

## Action Performed:
Precondition: Gmail account has a workspace connected to Xero.

1. Navigate to https://staging.new.expensify.com/
2. Log in with a Gmail account
3. Navigate to Workspace settings - Accounting - Import - Taxes
4. Enable "Import"
5. Wait for the sync to finish (do a manual sync if it didn't start)
6. Refresh the browser

## Expected Result:
"Taxes" should be available as soon as the sync is finished.

## Actual Result:
"Taxes" in the workspace settings are only appearing after a browser refresh.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b4f309df-adc0-4694-8686-34eb56343a66

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869826922075181653</li>
        <li>Upwork Job ID: 1869826922075181653</li>
        <li>Last Price Increase: 2025-01-02</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sobitneupane</details>

**Comment by CortneyOfstad** [comment]
Working on testing this — have to confirm if taxes are going to appear for US Xero connections as we usually turn that off by default

**Comment by CortneyOfstad** [comment]
Okay, got confirmation and was able to recreate the issue, so getting eyes on this

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"Taxes" in the workspace settings are only appearing after a browser refresh.

### What is the root cause of that problem?

- When we enable import taxes, the backend doesn't return the policy data to enable the tax features. After we reload the page, the policy data is updated with `trackingEnabled` as `true`
<img width="623" alt="Screenshot 2024-12-20 at 17 55 21" src="https://github.com/user-attachments/assets/dc158a6c-8b4d-4ed1-a0bf-3d78e1857da7" />

- It doesn't happen when we disable import taxes because the backend return the policy data that disable the tax feature
<img width="623" alt="Screenshot 2024-12-20 at 17 56 00" src="https://github.com/user-attachments/assets/74a11740-7078-4bdc-9f94-3a08f99cd9bf" />

### What changes do you think we should make in order to solve the problem?
1. The backend should return the policy data with `trackingEnabled` as `true` to enable the tax feature when we enable the taxes import

2. I think it's good to enable/disable the tax feature in optimistic data and reset it in failure data


https://github.com/Expensify/App/blob/78337bd1b2fe3c6b54cd151af8d36832f810e059/src/libs/actions/connections/Xero.ts#L340

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)



**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by CortneyOfstad** [comment]
Hey @sobitneupane! Just a heads up that, I will be OOO starting this afternoon (December 20th) and will be returning January 6th. A handful of folks on the BZ team will be online for a few days in between the 25th and the 1st, but we'll be operating with a skeleton crew. If any action is needed from a BZ perspective, please post this issue in #expensify-open-source and someone on the team will jump in. Otherwise I will review when I am back in office. 

Thank you and Happy Holidays!

**Comment by sobitneupane** [comment]
@mkzie2 I could not reproduce the issue. I connected Xero. > Import > Taxes > Enable Tax

Were you able to reproduce the issue? Am I missing any step?


https://github.com/user-attachments/assets/7d0ad28c-e527-4750-8a8c-c15b1666db63

**Comment by mkzie2** [comment]
Not able to reproduce now.

**Comment by sobitneupane** [comment]
Issue Not Reproducible.

**Comment by sobitneupane** [comment]
Issue Not Reproducible.

**Comment by sobitneupane** [comment]
@CortneyOfstad It seems that the issue is no longer reproducible.

**Comment by CortneyOfstad** [comment]
Yeah, I cannot recreate either, so going to close. Thanks @sobitneupane!
